File: CSharp/leetcodesln/SwapNodesInPairs.cs
namespace leetcodesln
{
    public class SwapNodesInPairs
    {
        public ListNode SwapPairs(ListNode head)
        {
            if (head == null || head.next == null) return head;

            var newHead = SwapPairs(head.next.next);
            var temp = head.val;
            head.val = head.next.val;
            head.next.val = temp;
            head.next.next = newHead;
            return head;
        }
    }
}

File: Go/questions/SwapNodesInPairs.go
package questions

import "github.com/leetcode/common"

func SwapPairs(head *common.ListNode) *common.ListNode {
	if head == nil || head.Next == nil {
		return head
	}

	newHead := SwapPairs(head.Next.Next)
	head.Val, head.Next.Val = head.Next.Val, head.Val
	head.Next.Next = newHead
	return head
}
